---
title: "Symbiodiniaceae *ITS2* sequencing library preparation"
author: "Ryan Eckert -- ryan.j.eckert@gmail.com"
output:
  html_document:
    theme: flatly
    code_folding: hide
    toc: yes
    toc_depth: 3
    toc_float: yes
    keep_md: no
  pdf_document:
    toc: yes
    toc_depth: '3'
---
```{r, setup, include = FALSE, message = FALSE, warning = FALSE}
knitr::opts_chunk$set(echo = FALSE)

if (!require("pacman")) install.packages("pacman")

pacman::p_load(kableExtra)

supplies = read.csv("supplies.csv")
ctab = read.csv("ctab.csv")
reagents = read.csv("reagents.csv")
etbrGel = read.csv("etbrGel.csv")
sybrGel = read.csv("sybrGel.csv")
barcodeMM = read.csv("barcodeMM.csv")
its2MM = read.csv("its2MM.csv")
its2PCR = read.csv("its2PCR.csv")
bcPCR = read.csv("bcPCR.csv")
its2Primers = read.csv("its2Primers.csv")
bcPrimers = read.csv("bcPrimers.csv")
```
#### version: `r Sys.Date() %>% format(format="%B %d, %Y")`  

#### [GitHub repository](https://github.com/RyanEckert/Belize_Mcav_Symbiodiniaceae_ITS2.git){target="_blank"}
***
# About this document
This protocol is used to extract high quality genomic DNA from both coral host and Symbiodiniaceae spp. The resulting DNA should be sufficient for downstream PCR and high-thoughout sequencing. This protocol was utilized to examine Symbiodiniaceae community structure across depth through *ITS2* metabarcoding in **Eckert RJ, Reaume A, Sturm AB, Studivan MS, and Voss JD.(in review) Symbiodiniaceae community variation among Montastraea cavernosa populations along a depth gradient on the Belize Barrier Reef.**   
    
Additionally, the DNA extraction and clean-up were used in [Eckert, R. J., Studivan, M. S., and Voss, J. D. (2019). Populations of the coral species *Montastraea cavernosa* on the Belize Barrier Reef lack vertical connectivity. Sci. Rep. 9, 7200.](https://doi.org/10.1038/s41598-019-43479-x){target="_blank"} to examine coral host population genetic structure.<br><br>

# CTAB gDNA Extraction Protocol 
***
Revised from [Mieog et al. (2009)](https://doi.org/10.1111/j.1755-0998.2008.02222.x){target="_blank"} and [http://ccb.ucr.edu/lab_protocols.html](http://ccb.ucr.edu/lab_protocols.html){target="_blank"}
<br>

## Supplies
The following supplies are necessary to perform a round of 24 extractions.  

```{r, supplies table}
kable(supplies) %>%
  kable_styling(bootstrap_options = c("condensed", "hover","striped"), full_width = FALSE, 
                position = 'left') %>%
  column_spec(1, width = "16em") %>%
  column_spec(2, width = "18em") %>%
  column_spec(3, width = "10em") %>%
  row_spec(2, hline_after = TRUE)
```

## Reagent recipes
Recipes for making ectraction buffer and associated stock solutions necessary for CTAB gDNA extraction follow.
<br>

### 2X CTAB extraction buffer
Use heat and stirring to dissolve CTAB into solution before adding NaCl. Make CTAB buffer just prior to extractions. Can keep at room temperature for several days.
```{r, ctab buffer, fig.align = 'left'}
kable(ctab, col.names = (c("Reagent","per 20 mL", "Target"))) %>%
  kable_styling(bootstrap_options = c("condensed", "hover","striped"), full_width = 
                FALSE, position = 'left') %>%
  column_spec(1, width = "33em") %>%
  column_spec(2, width = "10em")
```
  
### Reagent stock recipes             
```{r, reagent stocks}
kable(reagents, col.names = (c("Reagent","per 100 mL"))) %>%
    kable_styling(bootstrap_options = c("condensed", "hover","striped"), full_width 
                  = FALSE, position = 'left') %>%
  column_spec(1, width = "13em") %>%
  column_spec(2, width = "27em")
```
 
## Genomic DNA Extraction  
Set heatblock to 55 ºC  
Set refridgerated centrifuge and set to 4 ºC  
  
1. Prepare CTAB extraction buffer just prior to use, do not add proteinase K.
2. Scrape tissue from coral fragment and place into a 2 mL tube with 0.1 mL (~ 0.075 g) of 0.5 mm glass beads.
3. Add 800 µL CTAB extraction buffer.
4. Add 0.8 µL Proteinase K. Seal tubes with parafilm. Invert to mix. 
5. Bead beat for 2--3 mins (6 M/s, three 45 sec intervals w/ 2 min cool down between).
6. Incubate at 60 ºC for 90 min while mixing.
7. Add 800 µL Chloroform:Isoamyl  Alcohol (24:1). Invert to mix. 
8. Mix and centrifuge at 20,000 x *g* for 15 mins at 4 ºC.
9. Transfer aqueous phase to new 2 mL tube (600 µL then 150 µL), taking care not to disturb interphase layer. 
10. Add 800 µL cold (-20 ºC) Isopropanol.
11. Mix and incubate for 20 min at -20 ºC.
12. Centrifuge at 20,000 x *g* for 20 min at 4 ºC.
13. Carefully pour off supernatant. 
14. Add 150 µL of 70% Ethanol at room temperature. Invert to mix.
15. Centrifuge at 20,000 x *g* for 5 min at 4 ºC.
16. Remove supernatant (pour off, quick spin, pipette off remaining avoiding pellet) and dry inverted on Kimwipe for 15 min at room temperature. 
17. Elute in 100--200 µL of 1X TE pH 8.0.
18. Incubate at 55 ºC for 10 min.
<br><br>

# Cleaning genomic DNA
***
After extracting genomic DNA, use Zymo DNA Clean & Concentrator-5 (D4014) to clean DNA and remove inhibitors prior to running a PCR. This greatly improves nanodrop 230/260 and 260/280 readings, dramatically increasing amplification success.  

**Prior to first use add ethanol to Wash Buffer and label bottle.**  

1. Nanodrop DNA for concentration. Prepare 0.6 mL tubes with 5µg (or less) DNA in 100 µL total volume 1X TE to be cleaned.
2. Set Elution Buffer for elution step in heat block at 60--70 ºC.
3. In a 2 mL tube add 2:1 volume of Binding Buffer:DNA (200 µL) to each volume of genomic 
DNA and vortex thoroughly.
4. Transfer the mixture to a provided Zymo-Spin Column in a collection tube.
5. Centrifuge 10,000 x *g* for 30 sec at room temperature. Discard flow through.
6. Add 200μL DNA Wash Buffer to the column. Centrifuge at 10,000 x *g* for 1 min at room 
temperature. Repeat.
7. Transfer the column to a new labeled 1.5 mL tube. Elute by adding 20 µL of Elution 
Buffer directly to the column matrix and incubate at room temperature for 3–5 min. 
Centrifuge for 30 seconds to elute DNA.
8. Check DNA quality with Nanodrop and quantify DNA flourescently (e.g. Qubit) and prepare 10 ng/µL dilutions.
<br><br>

# Symbiodiniaceae *ITS2* amplification
***
Adapted from [Klepac et al. (2013)](https://doi.org/10.3354/meps11369){target="_blank"}. 
We want to try to avoid over-amplification, so we will start with 22 cycles and check on an agarose gel before final extension.  

Starting with **clean** DNA template that was quantified **fluorescently** will greatly increase amplification success.
<br>

## *ITS2* PCR
```{r, its2 master mix}
kable(its2MM, col.names = (c("Reagent","1 rxn"))) %>%
    kable_styling(bootstrap_options = c("condensed", "hover","striped"), full_width = FALSE,
                  position = 'float_left') %>%
  column_spec(1, width = "12em") %>%
  column_spec(2, width = "15em") %>%
  row_spec(8, bold = TRUE) %>%
  add_header_above(c("Master mix recipe"=2), line = FALSE, font_size = "Large", align = 'left')

kable(its2PCR, col.names = (c("","",""))) %>%
    kable_styling(bootstrap_options = c("condensed"), full_width = FALSE,
                  position = 'right') %>%
  column_spec(1:2, width = "5em") %>%
  row_spec(2:4, bold = TRUE, background = "#2C3E50", color = "white") %>%
  row_spec(0, bold = FALSE) %>%
  add_header_above(c("PCR profile"=3), line = FALSE, font_size = "Large", align = 'left')
```
<br><br><br><br><br><br> 

1. Amplify samples with the *ITS2* forward and reverse primers (see ITS2-F-miseq and ITS2-R-miseq under [Primer sequence information](#primer-sequence-information)) using cycle checks to obtain a faint but distinct band (Should take ~22 cycles). Avoid over-amplification, don't run  more than 28 cycles. To add cycles, place samples back into thermocycler and run for the additional number of cycles (no initial heating or final exstension steps).
2. Visualize on a gel using 3 µL of PCR product.  
      a. Use 3 µL PCR Product and 2 µL loading dye. Use 3 µL Ladder and 2 µL loading dye for marker wells.
      b. Run gel at 150 V for 15--25 min. You should see a distinct band at ~400 bp.
      c. If band is still not visible after checked on 3 gels, redo reaction using appropriate number of cycles.
3. Run a final extenstion on all samples.
3. Clean PCR product with geneJET PCR Purification Kit.  
<br>

```{r, etbr gel}
kable(etbrGel, col.names = (c("Reagent","For 300 mL gel"))) %>%
    kable_styling(bootstrap_options = c("condensed", "hover","striped"), full_width = FALSE,
                  position = 'left') %>%
  column_spec(1, width = "12em") %>%
  column_spec(2, width = "8em") %>%
  add_header_above(c("Sodium Borate/EtBr 1.5% Agarose gel Recipe"=2), line = FALSE, font_size = "Large") %>%
  add_footnote("Add EtBr just before pouring gel (60--55 ºC)", notation = "symbol")
```

## Cleaning PCR products  
Here we're cleaning with ThermoScientific geneJET.  
You can clean your PCR products with any commercially available PCR purification kit.
Zymo DCC-5 can also be used to clean PCR products by changing the binding buffer:DNA ratio to 5:1.  

**Prior to first use add ethanol to Wash Buffer and label bottle.**  

1. Add 1:1 volume of Binding Buffer:PCR product (27 µL). Mix thoroughly. Check color of solution after adding Binding Buffer. Yellow indicates optimal pH for DNA binding. If orange or violet, add 10 µL of 3M sodium acetate (pH 5.2) and mix.
2. Transfer reaction mixture/binding buffer solution to geneJET purification column.
3. Centrifuge at 12,000 x *g* for 1 min at room temperature. Discard flow-through.
4. Add 700 µL of Wash Buffer to purification column.
5. Centrifuge at 12,500 x *g* for 1 min at room temperature. Discard flow-through.
6. Centrifuge empty column for an additional 1 minute to completely remove Wash Buffer as residual ethanol may inhibit subsequent reactions.  
7. Transfer purification column to 1.5 mL microcentrifuge tube. Add desired volume of Elution Buffer.
      a. 30 µL used for PCR products after adding universal ITS2 primers with linkers. 
      b. 40 µL used for PCR products after adding barcode and MiSeq adapter primers.  
8. Allow to sit at room temperature for 1 minute, then centrifuge at 13,000 x *g* for 1 min at room temperature.  
9. Nanodrop cleaned sample and dilute with Elution Buffer for a final concentration of 10 ng/µL.
 <br><br>
 
# Symbiodiniaceae *ITS2* library barcoding
***
Now run a short PCR (4--6 cycles maximum) to incorporate a unique combination of indexed Illumina p5 and p7 adapters to each sample to pool for sequencing.
<br>

## Barcoding PCR                                    
```{r, barcode master mix}
kable(barcodeMM, col.names = (c("Reagent","1 rxn"))) %>%
    kable_styling(bootstrap_options = c("condensed", "hover","striped"), full_width = FALSE,
                  position = 'float_left') %>%
  column_spec(1, width = "10em") %>%
  column_spec(2, width = "16em") %>%
  row_spec(8, bold = TRUE) %>%
  add_header_above(c("Master mix recipe"=2), line = FALSE, font_size = "Large", align = 'left')

kable(bcPCR, col.names = (c("","",""))) %>%
    kable_styling(bootstrap_options = c("condensed"), full_width = FALSE,
                  position = 'right') %>%
  column_spec(1:2, width = "5em") %>%
  column_spec(3, width = "7em") %>%
  row_spec(2:4, bold = TRUE, background = "#2C3E50", color = "white") %>%
  row_spec(0, bold = FALSE) %>%
  add_header_above(c("PCR profile"=3), line = FALSE, font_size = "Large", align = 'left')
  
```  
<br><br><br><br><br><br>

1. Use cleaned and diluted samples in a PCR to incorporate barcoded MiSeq-adapter primers (see [Primer sequence information](#primer-sequence-information) for information on indexed primers).  
2. Load samples on 2% agarose gel stained with 30 µL SYBR green (1:10,000).
      a. Use all 20 µL of indexed PCR product on gel with 5 µL loading dye.
      b. Use 6 µL Ladder and 10 µL loading dye for marker wells.
3. Run gel at 80 V for ~90 min. This should give good separation of ladder and help with size selection.  
4. You should see a single band approximately 500 bp.
5. Excise DNA fragment from agarose gel with a clean, sharp scalpel. Avoid the edges of the band (i.e. take middle ~85 % of band). 
6. removing as much of the agarose on the gel slice surrounding the band as possible. Place into a 1.5 mL tube.
<br>
```{r, sybr gel}
kable(sybrGel, col.names = (c("Reagent","For 300 mL gel"))) %>%
    kable_styling(bootstrap_options = c("condensed", "hover","striped"), full_width = FALSE, 
                  position = 'left') %>%
  column_spec(1, width = "12em") %>%
  column_spec(2, width = "8em") %>%
  add_header_above(c("Sodium Borate/SYBR 2% Agarose gel Recipe"=2), line = FALSE, font_size = "Large", align = 'left') %>%
  add_footnote("Add SYBR green just before pouring gel (60--55 ºC)", notation = "symbol")
```

## QIAquick gel extraction
Here we used the QIAquick gel extraction kit. You can use any commercially available kit or alternatively do a "freeze and squeeze" gel extraction.  
  
**Make sure to add ethanol to Buffer PE before the first use**  
Set heat block to 50 ºC  
All centrifugation steps are at 17,900 x *g*

1. Add 3 volumes Buffer QG to 1 volume gel (100 mg ≈ 100 µL). For ITS2 barcoding size selection I found the average weight to be ~100 mg and used that as gel volume (i.e. add 300 µL Buffer QG).
2. Incubate at 50 ºC for 10 min (until gel slice has completely dissolved). Vortexing or mixing every 2--3 min helps dissolve the gel. 
3. After the gel slice has dissolved the mixture should be yellow in color. If orange or violet, add 10 µL 3M Sodium Acetate (pH 5.0) and mix. The mixture should turn yellow. (I’ve never had to perform this step).
4. Add 100 µL (i.e. 1 gel volume) Isopropanol to the sample and mix.
5. Apply the sample (up to 800 µL at a time) to a QIAquick spin column and provided catch tube. Centrifuge for 1 min. Discard flow through and place spin column back into the same catch tube.
6. Add 500 µL Buffer QG to the spin column and centrifuge for 1 min. Discard flow through and place spin column back into the same catch tube.
7. Add 750 µL Buffer PE to spin column and let stand for 2--5 min. Centrifuge for 1 min to wash DNA. Discard flow through and place spin column back into the same catch tube. Centrifuge empty column for 1 min to ensure all wash buffer has been removed. 
8. Place spin column into a clean, labeled 1.5 mL tube.
9. Add 30--50 µL of Buffer EB or nuclease-free water to the center of the spin column. Let incubate for 1--4 min. Elute DNA by centrifuging for 1 min.
<br><br>

# Library pooling
***
## qPCR library quantification
From [Eli Meyer](https://github.com/Eli-Meyer/ASV_utilities){target="_blank"}
<br>

1. Prepare a 1:100 dilution of each library by combining 2 µL of the eluted library with 198 µl NFW 
2. To quantify each library using qPCR, prepare a master mix. Complete a replicate of 2 wells per sample.
3. Pipette 13 µL qPCR master mix into each well of a PCR plate (Use Real-Time qPCR Plates and Optical Tape), then add 2 µL of each diluted library to the appropriate well.
4. Conduct qPCR using program ‘qPCR Quantification’ and calculate CT for each sample.
5. Repeat steps 2-4 with a second plate containing the second half of the samples.
6. To determine volumes of each library for the combined pool:
      a. Rank samples from lowest to highest CT and identify reference sample (sample with highest CT)
      b. Calculate proportion of each library to sequence as:
      $$P_L = 2^{[CT_{(sample)} – CT_{(reference)}]}$$  
      c. Calculate the volume of each library to use as:
      $$V = P_L * X~µL$$
          i. where $X$ = volume  of refernce sample to pool (20--30 µL)
      d. Note: If you’ve chosen a reference sample with a very high CT (suggesting a failed library prep) relative to the others, very low volumes (<2 µL) may be calculated at this step. If so, choose the next sample (i.e. next lowest CT) as reference instead, and continue adjusting choice of reference until reasonably high volumes are calculated for the majority of samples. 
      e. Note: As a rough rule of thumb, the pool of combined samples used for a single lane of sequencing should be at least 200-500 µL at this stage (it may be substantially higher). 
7. Combine libraries using the volumes calculated from qPCR to produce a pool for sequencing. 
<br>

## Pooled library concentration
Pooled libraries will likely need to be concentrated to send out for sequencing. Sequencing facilities typically require ≤20 μl volume at ≥2 nM concentration.
This can be done with a commercial PCR kit (e.g. Zymo DCC-5) following manufacturers protocol for cleaning PCR clean up.  
Additionally, it can be done with ethanol precipitation.
<br>

### Isopropanol precipitation 

1. Add 0.1 volumes of 3M sodium acetate and 3 volumes 100% isopropanol. 
2. Incubate 30 minutes at --20°C.
3. Centrifuge 20 minutes at maximum speed, 4°C.
4. Dry pellet 15 min at room temperature and resuspend in 25 μl NFW.
<br><br>

# *ITS2* amplicon analyses
***
All necessary data and code walkthroughs for [Symbiodinaceae *ITS2* amplicon analyses](https://ryaneckert.github.io/Belize_Mcav_Symbiodiniaceae_ITS2/stats){target="_blank"}
<br><br>

# Primer sequence information
***
This protocol was developed to reduce sequencing costs by giving each sample a unique “barcode” or index sequence so that samples can be pooled and run as a single sample on the MiSeq platform. 

The ITS2 forward and reverse primer sequences are universal ITS2 sequences ( [Pochon et al., 2001](doi.org/10.1007/s002270100674){target="_blank"}) that have been modified to include a linker and adapter that any of the barcode/index primers can then bind to, shown below:  

```{r, its2 primer sequences}
kable(its2Primers, col.names = (c("Primer","Sequence: 5'-Universal adapter + ITS2 F(or)R -3'"))) %>%
  kable_styling(bootstrap_options = c("condensed", "hover","striped"), full_width = FALSE, 
                position = 'left') %>%
  column_spec(1, width = "8em") %>%
  column_spec(2, width = "38em")
```  
    
Barcoded primers that also contain the Illumina adapter needed to bind to the flow cell of the MiSeq platform can then be added to the amplified products. Dual indexing (placing unique barcode sequences on both the forward and reverse primers is a cost-efficient way to include more samples while purchasing less barcode primers (i.e. 20 forward and 20 reverse barcode primers can label up to 400 unique samples). Primers used for this publication can be found on [github](https://github.com/RyanEckert/Belize_Mcav_Symbiodiniaceae_ITS2){target="_blank"}.  

```{r, bc primer sequences}
kable(bcPrimers, col.names = (c("Primer","Sequence: 5'-Illumina p5(or)p7 + Index + Universal adapter-3'"))) %>%
  kable_styling(bootstrap_options = c("condensed", "hover","striped"), full_width = FALSE, 
                position = 'left') %>%
  column_spec(1, width = "7em") %>%
  column_spec(2, width = "32em")
```

For more primer and barcode examples:<br>

* [https://wikis.utexas.edu/display/GSAF/Illumina+-+all+flavors](https://wikis.utexas.edu/display/GSAF/Illumina+-+all+flavors){target="_blank"}
* [https://wikis.utexas.edu/display/GSAF/rRNA+bacterial+gene+and+fungal+ITS+metagenomics+samples](https://wikis.utexas.edu/display/GSAF/rRNA+bacterial+gene+and+fungal+ITS+metagenomics+samples){target="_blank"}